### PR TITLE
#13087 make navtool work again

### PR DESCRIPTION
**Pull request by erickgonzalez**
To avoid duplicate items (same identifier diff language) now we check if the item exists in the language requested, if no we'll add the one in the default language, but if it exists we'll only add the one in the language requested.

### CODE REVIEW

**Review by jgambarios — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java [hunk 1]
```diff
@@ -296,6 +316,28 @@ private Host getHostFromPath(String path) throws DotDataException, DotSecurityEx
         return null;
     }
 
+    private boolean findHTMLPageInRequestedLanguage(List items, String identifier, long language){
```
[the remark this answers is not in the document]
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java]: let's rename this to `doesHTMLPageInRequestedLanguageExist` to be very clear

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java [hunk 2]
```diff
@@ -296,6 +316,28 @@ private Host getHostFromPath(String path) throws DotDataException, DotSecurityEx
...
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean findFileAssetInRequestedLanguage(List items, String identifier, long language){
+        for(Object item : items){
+            if(item instanceof IFileAsset){
+                if(((IFileAsset) item).getPermissionId().equals(identifier) && ((IFileAsset) item).getLanguageId() == language){
```
[the remark this answers is not in the document]
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java]: Same here `doesFileAssetInRequestedLanguageExist`

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java [hunk 3]
```diff
@@ -142,8 +148,15 @@ protected NavResultHydrated getNav(Host host, String path, long languageId, User
...
+                    boolean pageExistInRequestedLanguage = true;
 
-                    if (itemPage.getLanguageId() == languageId || LanguageWebAPI.canDefaultPageToDefaultLanguage()) {
+                    if(itemPage.getLanguageId() != languageId){
+                        pageExistInRequestedLanguage = findHTMLPageInRequestedLanguage(menuItems,itemPage.getIdentifier(),languageId);
+                    }
+
+
+                    if (itemPage.getLanguageId() == languageId ||
+                            (LanguageWebAPI.canDefaultPageToDefaultLanguage() && itemPage.getLanguageId() == defaultLanguage && !pageExistInRequestedLanguage)) {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java]: This is a very long second part of the OR. I would extract to a little method called `shouldAddPageInGivenLanguage(langId)`

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java [hunk 4]
```diff
@@ -194,8 +207,15 @@ protected NavResultHydrated getNav(Host host, String path, long languageId, User
...
+                    boolean fileExistInRequestedLanguage = true;
+
+                    if(itemFile.getLanguageId() != languageId){
+                        fileExistInRequestedLanguage = findFileAssetInRequestedLanguage(menuItems,itemFile.getPermissionId(),languageId);
+                    }
+
 
-                    if (itemFile.getLanguageId() == languageId || LanguageWebAPI.canDefaultFileToDefaultLanguage()) {
+                    if (itemFile.getLanguageId() == languageId ||
+                            (LanguageWebAPI.canDefaultFileToDefaultLanguage() && itemFile.getLanguageId() == defaultLanguage && !fileExistInRequestedLanguage)) {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java]: same as above, extract to little method the second part of the OR condition: shouldAddFileInGivenLanguage(langId)

**Review by nollymar — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**